actionview/app/assets/javascripts/rails-ujs/utils/csp.coffee

```diff
@@ -1,4 +1,7 @@
-# Content-Security-Policy nonce for inline scripts
-cspNonce = Rails.cspNonce = ->
-  meta = document.querySelector('meta[name=csp-nonce]')
-  meta and meta.content
+nonce = null
+
+Rails.loadCSPNonce = ->
+  nonce = document.querySelector("meta[name=csp-nonce]")?.content
+
+# Returns the Content-Security-Policy nonce for inline scripts.
+Rails.cspNonce = -> nonce
```

Comment:
Does the initial `nonce` need to be read on `DOMContentLoaded` or could the value be lazily read and cached?

```coffee
nonce = null
Rails.cspNonce = ->
  nonce ?= document.querySelector("meta[name=csp-nonce]")?.content
```

Reply:
> Does the initial nonce need to be read on `DOMContentLoaded`

Yes, since we may not call `cspNonce` until after the first page change.

Reply:
Gotcha. May want to ensure calling `cspNonce()` *before* DOMContentLoaded loaded works:
```coffee
Rails.cspNonce = -> 
  nonce ? Rails.loadCSPNonce()
```